backend/app/routers/datasets.py

```diff
@@ -358,7 +360,10 @@ async def edit_dataset(
     db: MongoClient = Depends(dependencies.get_db),
     user_id=Depends(get_user),
     es=Depends(dependencies.get_elasticsearchclient),
+    allow: bool = Depends(Authorization("editor")),
 ):
+    if not allow:
+        raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
```

Comment:
404 here intended? not 403?

Reply:
Which should these be? I can change them all to 403.